envoy/upstream/load_balancer.h

```diff
@@ -113,6 +142,11 @@ class LoadBalancerContext {
    * and return the corresponding host directly.
    */
   virtual absl::optional<OverrideHost> overrideHostToSelect() const PURE;
+
+  /* Called by the load balancer when asynchronous host selection completes
+   * @param host supplies the upstream host selected
+   */
+  virtual void onAsyncHostSelection(HostConstSharedPtr host) PURE;
```

Comment:
nit: AIUI, this will make a copy of the `shared_ptr` which can potentially be expensive since it has to increment the reference count. Maybe this could take `&&` to require it to be moved?

Reply:
done